catalog/core/catalog-core-standardframework/src/main/java/org/codice/ddf/catalog/resource/cache/impl/ResourceCacheService.java

```diff
@@ -0,0 +1,164 @@
+        this.mBeanServer = mBeanServer;
+        this.resourceCache = resourceCache;
+        this.downloadManager = downloadManager;
+        this.objectName = new ObjectName(OBJECT_NAME);
+        this.frameworkProperties = frameworkProperties;
+        this.catalogFramework = catalogFramework;
+    }
+    
+    public void init() {
+        registerMBean();
+        LOGGER.debug("Registered [{}] MBean under object name: [{}].", CLASS_NAME,
+                objectName.toString());
+    }
+
+    public void destroy() {
+        if (objectName != null && mBeanServer != null) {
+            unregisterMBean();
+            LOGGER.debug("Unregistered [{}] MBean.", CLASS_NAME);
+        }
+    }
+        
+    @Override
+    public boolean isCacheEnabled() {
+        return downloadManager.isCacheEnabled();
+    }
+
+    @Override
+    public boolean contains(Metacard metacard) {
+        return downloadManager.isCacheEnabled() ? resourceCache.contains(metacard) : false;
+    }
+
+    @Override
+    public boolean containsById(String metacardId) {
+        Optional<Metacard> optionalMetacard = queryForMetacard(metacardId);
+        return downloadManager.isCacheEnabled() && optionalMetacard.isPresent()
+                ? contains(optionalMetacard.get()) : false;
+    }
+
+    private Optional<Metacard> queryForMetacard(String metacardId) {
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .attribute(Metacard.ID)
+                .is()
+                .equalTo()
+                .text(metacardId);
+        QueryRequest queryRequest = new QueryRequestImpl(new QueryImpl(filter), true);
+        QueryResponse queryResponse = null;
+        try {
+            queryResponse = catalogFramework.query(queryRequest);
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            LOGGER.error("Unable to lookup metacard for metacard id [{}].", metacardId);
+            return Optional.empty();
+        }
+        return queryResponse != null && queryResponse.getResults()
+                .size() == 1 ? Optional.of(queryResponse.getResults().get(0).getMetacard()) : Optional.empty();
+    }
+
+    private void registerMBean() {
+        if (!mBeanServer.isRegistered(objectName)) {
+            try {
+                // wrap the implementation in a StandardMBean so we can have the MBean interface
+                // and implementation in different packages
+                mBeanServer.registerMBean(new StandardMBean(this, ResourceCacheServiceMBean.class),
+                        objectName);
+            } catch (InstanceAlreadyExistsException | MBeanRegistrationException
+                    | NotCompliantMBeanException e) {
+                String message = String.format(
+                        "Unable to register [%s] MBean under object name: [%s].", CLASS_NAME,
+                        objectName.toString());
+                LOGGER.debug(message, e);
```

Comment:
Do we really need to log and throw? Are these not getting caught anywhere else in the system?